app/models/miq_request_workflow.rb

```diff
@@ -501,6 +501,16 @@ def required_description(dlg, fld)
     "'#{dlg[:description]}/#{fld[:required_description] || fld[:description]}'"
   end
 
+  def allowed_storage_profiles(options = {})
+    model_name = options[:category]
+    return @filters[model_name] unless @filters[model_name].nil?
+    template = VmOrTemplate.find_by_id(get_value(@values[:src_vm_id]))
```

Comment:
Use the `resources_for_ui` method which will return an empty hash if auto_placement is enabled.

[app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb#L117](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb#L117)

You also need to handle the source vm being `nil` which happens when the provisioning workflow is created as part of a new service catalog item.

Finally, with the service catalog item process the source VM can change and you will need to clear the cached values when that happens.  You will want to look at overriding the `set_on_vm_id_changed` method at the VMware workflow subclass to do this.